slsDetectorSoftware/src/DetectorImpl.cpp

```diff
@@ -1242,25 +1242,45 @@ int DetectorImpl::acquire() {
     return OK;
 }
 
-void DetectorImpl::startAcquisition(bool blocking, std::vector<int> positions) {
-    // handle Mythen3 synchronization
-    if (shm()->detType == defs::MYTHEN3 && size() > 1) {
-        std::vector<int> master;
-        std::vector<int> slaves;
-        if (positions.empty() ||
-            (positions.size() == 1 && positions[0] == -1)) {
-            positions.resize(modules.size());
-            std::iota(begin(positions), end(positions), 0);
+void DetectorImpl::getMasterSlaveList(std::vector<int> positions,
+                                      std::vector<int> &master,
+                                      std::vector<int> &slaves) {
+    // expand positions list
+    if (positions.empty() || (positions.size() == 1 && positions[0] == -1)) {
+        positions.resize(modules.size());
+        std::iota(begin(positions), end(positions), 0);
+    }
+    // could be all slaves in positions
+    slaves.reserve(positions.size());
+    auto is_master = Parallel(&Module::isMaster, positions);
+    for (size_t i : positions) {
+        if (is_master[i])
+            master.push_back(i);
+        else
+            slaves.push_back(i);
+    }
+}
+
+void DetectorImpl::startAcquisition(const bool blocking,
+                                    std::vector<int> positions) {
+    bool handleSynchronization = false;
+    // multi module m3 or multi module sync enabled jungfrau
+    if (size() > 1) {
```

Comment:
Is there a way around having to differentiate between sometimes M3 sometimes not?

Reply:
Only the software trigger is not implemented for M3.,. so..its either multi module m3 or its not..
but we also check for multi module jungfrau that has sync enabled..